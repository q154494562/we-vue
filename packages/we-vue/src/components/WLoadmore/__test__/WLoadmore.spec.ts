import { shallowMount } from '@vue/test-utils'
import Loadmore from '../'

describe('loadmore', () => {
  test('create', () => {
    const wrapper = shallowMount(Loadmore, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-loadmore')
    expect(wrapper.classes()).toContain('weui-loadmore')
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('text', () => {
    const wrapper = shallowMount(Loadmore, {
      propsData: {
        text: 'test',
      },
    })

    expect(wrapper.find('.weui-loadmore__tips').text()).toBe('test')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
