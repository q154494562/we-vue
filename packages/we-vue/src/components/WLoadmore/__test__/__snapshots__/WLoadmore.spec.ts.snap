// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadmore create 1`] = `
<div class="weui-loadmore">
  <vuecomponent-stub type="default" size="17" color="#aaa"></vuecomponent-stub><span class="weui-loadmore__tips">正在加载</span>
</div>
`;

exports[`loadmore text 1`] = `
<div class="weui-loadmore">
  <vuecomponent-stub type="default" size="17" color="#aaa"></vuecomponent-stub><span class="weui-loadmore__tips">test</span>
</div>
`;
